zulipterminal/model.py

```diff
@@ -906,11 +907,35 @@ def _handle_typing_event(self, event: Event) -> None:
                 else:
                     raise RuntimeError("Unknown typing event operation")
 
-    def get_invalid_recipient_emails(self, recipient_emails: List[str]
-                                     ) -> List[str]:
-
-        return [email for email in recipient_emails
-                if email not in self.user_dict]
+    def get_invalid_recipients(self, recipient_text: str) -> List[str]:
+
+        recipients = [recipient.strip()
+                      for recipient in recipient_text.split(',')]
+        invalid_recipients = []
+
+        for recipient in recipients:
+            # Allow empty recipients.
+            if recipient:
+                email_match = re.findall(r'[\w\.-]+@[\w\.-]+', recipient)
+                if email_match:
+                    # Validate only the first email for a recipient.
+                    recipient_email = email_match[0]
+                    if recipient_email in self.user_dict:
+                        cleaned_recipient = (recipient.replace("<", "")
+                                             .replace(">", ""))
+                        recipient_name, extra_text = (cleaned_recipient
+                                                      .split(recipient_email))
+
+                        # Ensure that there is no text after the email
+                        # (extra_text) and that the current name matches
+                        # the actual name of the recipient.
+                        if not extra_text.strip() and (recipient_name.strip()
+                           == self.user_dict[recipient_email]['full_name']):
+                            continue
+
+                invalid_recipients.append(recipient)
```

Comment:
The code design here feels strange - 4 nested if's, with a continue, or else it appends. Could you use a more complex regex to avoid all the manual cleaning?

Also the function of the code is different now - it is connected to the structure of the UI display. Ideally the model would simply validate name/email pairs passed in, with the caller doing any parsing into that format.

Reply:
I'll try using a more complex regex, I was a little uncertain about the design, and hence I pushed the commit as WIP.